include/grpc++/alarm.h

```diff
@@ -54,7 +56,11 @@ class Alarm : private GrpcLibrary {
   /// Once the alarm expires (at \a deadline) or it's cancelled (see \a Cancel),
   /// an event with tag \a tag will be added to \a cq. If the alarm expired, the
   /// event's success bit will be true, false otherwise (ie, upon cancellation).
-  Alarm(CompletionQueue* cq, gpr_timespec deadline, void* tag);
+  template <typename T>
+  Alarm(CompletionQueue* cq, const T& deadline, void* tag)
+      : tag_(tag),
+        alarm_(grpc_alarm_create(cq->cq(), TimePoint<T>(deadline).raw_time(),
+                                 static_cast<void*>(&tag_))) {}
 
```

Comment:
Hmm we can't use std::chrono because that results in a REALTIME clock, something not allowed in alarms (racy, see https://github.com/grpc/grpc/blob/master/src/core/iomgr/timer.c#L178). We could try to figure out how to guarantee a monotonic conversion from std::chrono::*, but that's a story for a different day.